src/allotropy/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_parser.py

```diff
@@ -191,8 +191,12 @@ def get_device_control_document_item(
         return DeviceControlDocumentItem(
             device_type="qPCR",
             measurement_method_identifier=data.header.measurement_method_identifier,
-            total_cycle_number_setting=TQuantityValueNumber(
-                value=well_item.amplification_data.total_cycle_number_setting,
+            total_cycle_number_setting=(
+                TQuantityValueNumber(
+                    value=well_item.amplification_data.total_cycle_number_setting,
+                )
+                if well_item.amplification_data is not None
```

Comment:
Should this use the `quantity_or_none` helper or some thing? Apologies the name isnt coming to mind but I can find it if you dont happen to know off hand

Reply:
I can't use quantity_or_none because the function expects a value that can be none. 
The problem is the nullable value is not total_cycle_number_setting but amplification_data.

So if I do something like 
**quantity_or_none(TQuantityValueNumber, well_item.amplification_data.total_cycle_number_setting)**
and amplification_data is none I will get an error saying None value has no attribute total_cycle_number_setting

So I have to check if amplification is none anyway

Reply:
Ah got it, thanks for the clarification